aws/sdk-codegen/src/main/kotlin/software/amazon/smithy/rustsdk/customize/s3/S3ExtendedRequestIdDecorator.kt

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.rustsdk.customize.s3
+
+import software.amazon.smithy.rust.codegen.client.smithy.ClientCodegenContext
+import software.amazon.smithy.rust.codegen.core.smithy.RuntimeType
+import software.amazon.smithy.rustsdk.BaseRequestIdDecorator
+import software.amazon.smithy.rustsdk.InlineAwsDependency
+
+class S3ExtendedRequestIdDecorator : BaseRequestIdDecorator() {
```

Comment:
Thinking about this more, I wonder if we should just code generate the `RequestId` trait entirely so that for S3, it can have two methods on it instead of one. Then there wouldn't be a need for two separate traits in S3 to access the two request IDs, and then duplicated code could be eliminated.

Reply:
This would benefit the S3 SDK, but would prevent accessing request IDs for types across SDK crates.

Reply:
can we have a root trait, then a second trait for S3 that gets re-exported?